Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -864,15 +824,19 @@ extension PurchasesOrchestrator: StoreKit2TransactionListenerDelegate {
         _ listener: StoreKit2TransactionListener,
         updatedTransaction transaction: StoreTransactionType
     ) async throws {
-        let isRestore = self.systemInfo.observerMode
-
-        _ = try await self.syncPurchases(receiptRefreshPolicy: .always,
-                                         isRestore: isRestore,
-                                         initiationSource: .queue)
+        let storefront = await Storefront.currentStorefront
 
-        await Async.call { completion in
-            self.finishTransactionIfNeeded(transaction) { @MainActor in
-                completion(())
+        _ = try await Async.call { completed in
+            self.transactionPoster.handlePurchasedTransaction(
+                StoreTransaction.from(transaction: transaction),
+                presentedOfferingID: nil,
+                storefront: storefront,
+                source: .init(
+                    isRestore: self.allowSharingAppStoreAccount,
+                    initiationSource: .queue
+                )
+            ) { _, customerInfo, error, _ in
+                completed(Result(customerInfo, error))
```

Comment:
Good point. I think it's still useful to make this method `wait` for this to finish, but we don't actually care about the result so I'll remove that.

Reply:
Actually this is used because of the `try`. This becomes a bit simpler in #2542:
```swift
_ = try await Async.call { completed in
    self.transactionPoster.handlePurchasedTransaction(
        StoreTransaction.from(transaction: transaction),
        data: .init(
            appUserID: self.appUserID,
            presentedOfferingID: nil,
            unsyncedAttributes: self.unsyncedAttributes,
            storefront: storefront,
            source: .init(
                isRestore: self.allowSharingAppStoreAccount,
                initiationSource: .queue
            )
        )
    ) { result in
        completed(result.mapError(\.asPurchasesError))
    }
}
```